src/components/Navbar.tsx

```diff
@@ -0,0 +1,183 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+import { IconDefinition } from "@fortawesome/free-regular-svg-icons";
+import { faCircle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import style from "../variables.module.scss";
+import "./navbar.scss";
+import { ThemeContext, ThemeProvider } from "../theme";
+import { isNavbarItem } from "../utils";
+
+const NavbarContext = React.createContext<{
+  selectedId?: number;
+  setSelectedId?: (id?: number) => void;
+  registerItem?: (item: React.RefObject<HTMLLIElement>) => void;
+}>({});
+
+type NavbarItemProps = {
+  icon: IconDefinition;
+  title: string;
+  onSelect: ({ title, id }: { title: string; id?: number }) => void;
+  id?: number; // TODO: find out how not to expose this prop
+};
+
+export function NavbarItem({ icon, title, onSelect, id }: NavbarItemProps) {
+  const theme = useContext(ThemeContext);
+  const { selectedId, setSelectedId, registerItem } = useContext(NavbarContext);
+  const itemRef = useRef<HTMLLIElement>(null);
+
+  const handleClick = () => {
+    // If is already selected, then deselect
+    if (id === selectedId) {
+      setSelectedId?.(undefined);
+    } else {
+      setSelectedId?.(id);
+      onSelect({ title, id });
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    switch (event.key) {
+      case "Enter":
+      case " ":
+        // If is already selected, then deselect
+        if (id === selectedId) {
+          setSelectedId?.(undefined);
+        } else {
+          setSelectedId?.(id);
+          onSelect({ title, id });
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    registerItem?.(itemRef);
+  }, [registerItem]);
+
+  const isSelected = id === selectedId;
+
+  return (
+    <li
+      tabIndex={0}
```

Comment:
question: Tell me, why do we need this `tabIndex`? What's the downside?

Hint: It would be better using `li > button`, or, `li > a` as these are accessible by default.

Reply:
changed it to use button for accessibility :accessibility:

Reply:
Great! **Now the next challenge:** What if a dev uses the component where one of the items goes to another page. How do you adapt the component to support both cases (button or link)?

Reply:
I went for using the `as` prop to pass `a` or `button` (with button as default)